File: src/main/java/life/magpies/registry/commands/CommandKit.java
package life.magpies.registry.commands;

import com.mojang.brigadier.context.CommandContext;
import net.fabricmc.fabric.api.command.v2.CommandRegistrationCallback;
import net.minecraft.entity.ItemEntity;
import net.minecraft.server.command.ServerCommandSource;
import net.minecraft.server.network.ServerPlayerEntity;
import net.minecraft.text.Text;
import net.minecraft.util.math.Box;
import net.minecraft.util.math.Vec3d;
import net.minecraft.world.World;

import static com.mojang.brigadier.arguments.IntegerArgumentType.getInteger;
import static com.mojang.brigadier.arguments.IntegerArgumentType.integer;
import static net.minecraft.server.command.CommandManager.argument;
import static net.minecraft.server.command.CommandManager.literal;

public class CommandKit {
    public static void register() {
        CommandRegistrationCallback.EVENT.register((dispatcher, registryAccess, environment) -> {
            dispatcher.register(literal("kit")
                    .executes(context -> execute(context, 5))
                    .then(argument("radius", integer())
                            .executes(ctx -> execute(ctx, getInteger(ctx, "radius")))
                    ));
        });
    }

    private static int execute(CommandContext<ServerCommandSource> context, int radius) {
        if (radius > 30) {
            context.getSource().sendFeedback(() -> Text.literal(("只支持到最大30半径")), false);
            return 0;
        }
        // 获取玩家的对象
        ServerPlayerEntity player = context.getSource().getPlayer();
        // 获取玩家的位置
        assert player != null;
        Vec3d playPos = player.getPos();
        // 获取世界对象
        World world = player.getEntityWorld();
        var ref = new Object() {
            int killEntities = 0;
        };
        // 遍历世界中的实体
        // 检查实体是否不是玩家本身
        // 检查实体是否是生物（如生物实体而不是方块或物品）
        world.getEntitiesByClass(ItemEntity.class, new Box(playPos.getX() - radius,
                        playPos.getY() - radius,
                        playPos.getZ() - radius,
                        playPos.getX() + radius,
                        playPos.getY() + radius,
                        playPos.getZ() + radius),
                (entity) -> true).stream().filter(entity -> !entity.equals(player)).forEach(entity -> {
            entity.kill();
            ref.killEntities++;
        });
        context.getSource().sendFeedback(() -> Text.literal("清理了附近的%s个掉落物".formatted(ref.killEntities)), false);
        return 1;
    }

}

File: src/main/kotlin/life/magpies/registry/commands/CommandKit.kt
package life.magpies.registry.commands

import com.mojang.brigadier.CommandDispatcher
import com.mojang.brigadier.arguments.IntegerArgumentType
import com.mojang.brigadier.context.CommandContext
import life.magpies.registry.config.ModConfig.kit
import life.magpies.registry.config.ModConfig.kit_Radius

import net.fabricmc.fabric.api.command.v2.CommandRegistrationCallback
import net.minecraft.command.CommandRegistryAccess
import net.minecraft.entity.ItemEntity
import net.minecraft.server.command.CommandManager
import net.minecraft.server.command.CommandManager.RegistrationEnvironment
import net.minecraft.server.command.ServerCommandSource
import net.minecraft.text.Text
import net.minecraft.util.math.Box

object CommandKit {
    fun register() {
        CommandRegistrationCallback.EVENT.register(CommandRegistrationCallback { dispatcher: CommandDispatcher<ServerCommandSource?>, registryAccess: CommandRegistryAccess?, environment: RegistrationEnvironment? ->
            dispatcher.register(
                CommandManager.literal("kit")
                    .requires { source: ServerCommandSource ->
                        source.hasPermissionLevel(
                            kit
                        )
                    }
                    .executes { context: CommandContext<ServerCommandSource> ->
                        execute(
                            context,
                            5
                        )
                    }
                    .then(
                        CommandManager.argument("radius", IntegerArgumentType.integer())
                            .executes { ctx: CommandContext<ServerCommandSource> ->
                                execute(
                                    ctx,
                                    IntegerArgumentType.getInteger(ctx, "radius")
                                )
                            }
                    )
            )
        })
    }

    private fun execute(ctx: CommandContext<ServerCommandSource>, radius: Int): Int {
        if (radius > kit_Radius) {
            ctx.source.sendFeedback({
                Text.of((("超过最大限制范围 $kit_Radius")))
            }, false)
            return 0
        }
        // 获取玩家的对象
        val player = CommandUtils.getPlayer(ctx) ?: return 0
        val playPos = player.pos
        // 获取世界对象
        val world = player.entityWorld
        var killEntities = 0
        // 遍历世界中的实体
        // 检查实体是否不是玩家本身
        // 检查实体是否是生物（如生物实体而不是方块或物品）
        world.getEntitiesByClass(
            ItemEntity::class.java, Box(
                playPos.getX() - radius,
                playPos.getY() - radius,
                playPos.getZ() - radius,
                playPos.getX() + radius,
                playPos.getY() + radius,
                playPos.getZ() + radius
            )
        ) { entity: ItemEntity? -> true }.stream().filter { entity: ItemEntity -> entity != player }
            .forEach { entity: ItemEntity ->
                entity.kill()
                killEntities++
            }
        ctx.source.sendFeedback({
            Text.literal(
                "清理了附近的${killEntities}个掉落物"
            )
        }, false)
        return 1
    }
}